src/terms/hubbard.jl

```diff
@@ -0,0 +1,268 @@
+        i = 1
+        while i <= nprojs
+            il = labels[i].l
+            if !(manifold_atoms[iatom] == labels[i].iatom)
+                i += 2*il + 1
+                continue
+            end
+            for jatom in eachindex(manifold_atoms)
+                j = 1
+                while j <= nprojs
+                    jl = labels[j].l
+                    (manifold_atoms[jatom] == labels[j].iatom) && (n_IJ[σ, iatom, jatom] = n_matrix[σ, i:i+2*il, j:j+2*jl])
+                    j += 2*jl + 1
+                end
+            end
+            for (ik, projk) in enumerate(projectors)
+                p_I[ik][iatom] = projk[:, i:i+2*il]  
+            end
+            i += 2*il + 1
+        end
+    end
+
+    n_IJ = symmetrize_nhub(n_IJ, basis.model.lattice, basis.symmetries, basis.model.positions)
+
+    return (; n_IJ=n_IJ, manifold_labels=labels, p_I=p_I)
+end
+
+function reshape_hubbard_proj(projectors::Vector{Matrix{Complex{T}}}, labels) where {T}
+    natoms = max([labels[i].iatom for i in 1:length(labels)]...)
+    p_I = [Vector{Matrix{Complex{T}}}(undef, natoms) for i in 1:length(projectors)]
+    i = 1
+    while i <= size(projectors, 2)
+        il = labels[i].l
+        iatom = labels[i].iatom
+        for (ik, projk) in enumerate(projectors)
+            p_I[ik][iatom] = Matrix{Complex{T}}(undef, size(projk,1), 2*il + 1)
+            p_I[ik][iatom] = copy(projk[:, i:i+2*il])
+        end
+        i += 2*il + 1
+    end
+
+    return p_I
+end
+
+struct Hubbard
+    manifold :: OrbitalManifold
+    U        :: Real
+end
+function (hubbard::Hubbard)(basis::AbstractBasis)
+    isempty(hubbard.U) && return TermNoop()
+    projs, labs = atomic_orbital_projectors(basis)
+    labels, projectors = extract_manifold(basis, projs, labs, hubbard.manifold)
+    U = austrip(hubbard.U)
+    TermHubbard(hubbard.manifold, U, projectors, labels)
+end
+
+struct TermHubbard{PT, L} <: Term
+    manifold :: OrbitalManifold
+    U        :: Real
+    P        :: PT
+    labels   :: L
+end
+
+@timing "ene_ops: hubbard" function ene_ops(term::TermHubbard,
+                                            basis::PlaneWaveBasis{T},
+                                            ψ, occupation; n_hub=nothing, ψ_hub=nothing,
+                                            labels=term.labels,
+                                            kwargs...) where {T}
+    if isnothing(ψ)
+        if isnothing(n_hub)
+           return (; E=zero(T), ops=[NoopOperator(basis, kpt) for kpt in basis.kpoints])
+        end
+        n = n_hub
+        ψ = ψ_hub
+        proj = reshape_hubbard_proj(term.P, term.labels)
+    else
+        Hubbard = compute_hubbard_nIJ(term.manifold, basis, ψ, occupation; projectors=term.P, 
+        labels)
+        n = Hubbard.n_IJ
+        n_hub = n
+        proj = Hubbard.p_I
+    end
+    to_unit = ustrip(auconvert(u"eV", 1.0))
+    U = term.U / to_unit  # U is usally expressed in eV in the literature, but we work in au
```

Comment:
This shouldn't be here anymore now that the `U` inside `TermHubbard` is converted to atomic units.